### PR TITLE
Fix cox3d_learner ONNX support

### PR DESCRIPTION
**Pull request by LukasHedegaard**
This PR is an implementation overhaul for the Continual X3D learner, which uses the continual-inference library to create proper ONNX export and inference for step-wise model operation.

### CODE REVIEW

File: src/opendr/perception/activity_recognition/x3d/x3d_learner.py [hunk 1]
```diff
@@ -116,7 +115,7 @@ def __init__(
         self.seed = seed
         self.num_classes = num_classes
         self.loss = loss
-        self.ort_session = None
+        self._ort_session = None
```
Comment (tsampazk) [inline src/opendr/perception/activity_recognition/x3d/x3d_learner.py:118]: On unchanged line 129 below, "`self.backbon`e" ->  "`self.backbone` "

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ Released on December, XX, 2022.
     - Fixed `BoundingBoxList`, `TrackingAnnotationList`, `BoundingBoxList3D` and `TrackingAnnotationList3D` confidence warnings ([#365](https://github.com/opendr-eu/opendr/pull/365)).
     - Fixed undefined `image_id` and `segmentation` for COCO `BoundingBoxList` ([#365](https://github.com/opendr-eu/opendr/pull/365)).
 
+  - Bug Fixes:
+    - Fixed Continual X3D ONNX support.
```
Comment (ad-daniel) [inline CHANGELOG.md:15]:
```suggestion
    - Fixed Continual X3D ONNX support ([#372](https://github.com/opendr-eu/opendr/pull/372)).
```

**Comment by ad-daniel** [comment]
> I assume that files in cox3d/algorithm, are used as-is and i only added a couple of comments for typos and skipped some other issues in the docstrings and function naming.

Yes I think we don't expect the algorithms to follow all the constraints we impose for our tools, especially considering that these algorithms might evolve so it might end up just complicating keeping them up to date. As long as the test sources passes (or the location is skipped when running the test) it should be fine IMHO.

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Just a detail, thank you

File: src/opendr/perception/activity_recognition/x3d/x3d_learner.py [hunk 3]
```diff
@@ -484,8 +475,8 @@ def infer(self, batch: Union[Video, List[Video], torch.Tensor]) -> List[Category
             batch = torch.stack([torch.tensor(v.data) for v in batch])
 
         batch = batch.to(device=self.device, dtype=torch.float)
-        if self.ort_session is not None:
-            results = torch.tensor(self.ort_session.run(None, {"video": batch.cpu().numpy()})[0])
+        if self._ort_session is not None:
+            results = torch.tensor(self._ort_session.run(None, {"video": batch.cpu().numpy()})[0])
         else:
             self.model.eval()
             results = self.model.forward(batch)
```
[the remark this answers is not in the document]
Comment (LukasHedegaard) [reply src/opendr/perception/activity_recognition/x3d/x3d_learner.py:482]: Fixed in cce42a75

File: src/opendr/perception/activity_recognition/x3d/x3d_learner.py (hunk 1 above)
Comment (LukasHedegaard) [reply src/opendr/perception/activity_recognition/x3d/x3d_learner.py:118]: Fixed in cce42a75

**Review by ad-daniel — APPROVED** [review]
Thank you!

**Review by tsampazk — APPROVED** [review]
Thank you!

**Pull request merged**